Fix #7: Unify class and interface naming to not use "JsonForms" as a prefix

lucas-koehler opened the pull request:
* Rename JsonFormsTreeWidget to MasterTreeWidget
* Rename tree props and internal function in util.ts
* JsonFormsTreeAnchor -> TreeAnchor
* JsonFormsTreeContextMenu -> TreeContextMenu
* Adapt the context menu path constants
* JsonFormsTreeEditorContribution -> BaseTreeEditorContribution
* JSONFormsWidget -> DetailFormWidget
* Rename css classes to use prefix 'theia-tree-editor'
* Remove unnecessary css class assignments in the tree widget
* Increase version to 0.6

eneufeld commented:
Generic comment: I suggest to not use a reference to the Issue in the commit header, only in the commit body. I like `Unify class and interface naming to not use "JsonForms" as a prefix`

eneufeld commented on src/browser/tree-widget.tsx:
```diff
@@ -12,7 +12,7 @@ import { Emitter, MenuPath } from '@theia/core';
 import { ContextMenuRenderer, RenderContextMenuOptions } from '@theia/core/lib/browser/context-menu-renderer';
 import { TreeNode } from '@theia/core/lib/browser/tree/tree';
-import { NodeProps, TreeProps, TreeWidget } from '@theia/core/lib/browser/tree/tree-widget';
+import { NodeProps, TreeProps, TreeWidget as TheiaTreeWidget} from '@theia/core/lib/browser/tree/tree-widget';
```
shouldn't we name our treewidget something like: `FormTreeWidget` or `MasterTreeWidget` to not conflict with theia?
We are not exporting our treewidget as API anyway, right?

lucas-koehler replied:
We are exporting it because users need to inject it in their constructors. But it's not extended by users.

I'm not sure what to prefer:

Generally, I don't think users ever need to import Theia's TreeWidget and ours in the same class. So the conflict shouldn't really be a problem. Because we never use the term tree-master-detail, and I don't know how common this term is outside our bubble, it might not be too intuitive to have a `MasterTreeWidget` and `FormTreeWidget` doesn't really fit imo because the tree is not part of the form.

On the other hand, I agree that naming conflicts are not too nice and also could lead to confusion.

I can rename it and in this case I'd chose `MasterTreeWidget` over `FormTreeWidget`.
What do you think?

planger replied:
:+1: I'm also for renaming it to avoid the name conflict. I would prefer `MasterTreeWidget` as it also specifies the role of tree in the context of the master detail editor. In the same line, we should maybe call the `FormWidget` rather `DetailFormWidget`?

lucas-koehler replied:
That sounds like a good plan. Together with some hint in the documentation what master detail means this should make it quite understandable. Thanks for the input :)

lucas-koehler commented:
@eneufeld I renamed `TreeWidget` to `MasterTreeWidget` and `FormWidget` to `DetailFormWidget`. I also squashed the changes and adapted the commit title.
